src/WebJobs.Script/Host/ScriptHostIdProvider.cs

```diff
@@ -8,36 +8,38 @@
 using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Host.Executors;
 using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Configuration;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.WebJobs.Script
 {
     public class ScriptHostIdProvider : IHostIdProvider
     {
+        private const string HostIdPath = ConfigurationSectionNames.JobHost + ":id";
         private readonly IConfiguration _config;
-        private readonly ScriptSettingsManager _settingsManager;
+        private readonly IEnvironment _environment;
         private readonly ScriptJobHostOptions _options;
 
-        public ScriptHostIdProvider(IConfiguration config, ScriptSettingsManager settingsManager, IOptions<ScriptJobHostOptions> options)
+        public ScriptHostIdProvider(IConfiguration config, IEnvironment environment, IOptions<ScriptJobHostOptions> options)
         {
             _config = config;
-            _settingsManager = settingsManager;
+            _environment = environment;
             _options = options.Value;
         }
 
         public Task<string> GetHostIdAsync(CancellationToken cancellationToken)
         {
-            return Task.FromResult(_config["id"] ?? GetDefaultHostId(_settingsManager, _options));
+            return Task.FromResult(_config[HostIdPath] ?? GetDefaultHostId(_environment, _options));
```

Comment:
This is one of the key changes... getting the host id with the appropriate configuration path. A test has been added to validate.